### PR TITLE
GH-116218 Docs: Add availability information for the 'resource' module

### PR DESCRIPTION
**Pull request by Privat33r-dev**
* Issue: gh-116218

----
📚 Documentation preview 📚: https://cpython-previews--116256.org.readthedocs.build/

### CODE REVIEW

File: Doc/library/resource.rst [hunk 1]
```diff
@@ -123,6 +123,8 @@ platform.
    This may result in the creation of a partial core file if a larger core would be
    required to contain the entire process image.
 
+   .. availability:: Linux >= 1.3.0
```
Comment (ronaldoussoren) [inline Doc/library/resource.rst]: This constant is part of the unix standard and should be available on all platforms (given that this module is documented as available on unix platforms).

File: Doc/library/resource.rst [hunk 2]
```diff
@@ -131,37 +133,51 @@ platform.
    the :mod:`signal` module documentation for information about how to catch this
    signal and do something useful, e.g. flush open files to disk.)
 
+   .. availability:: Linux >= 1.3.0
```
Comment (ronaldoussoren) [inline Doc/library/resource.rst]: Likewise.

File: Doc/library/resource.rst [hunk 3]
```diff
@@ -131,37 +133,51 @@ platform.
...
+
 
 .. data:: RLIMIT_FSIZE
 
    The maximum size of a file which the process may create.
 
+   .. availability:: Linux >= 1.3.0
```
Comment (ronaldoussoren) [inline Doc/library/resource.rst]: Likewise.

File: Doc/library/resource.rst [hunk 4]
```diff
@@ -131,37 +133,51 @@ platform.
...
+
 
 .. data:: RLIMIT_DATA
 
    The maximum size (in bytes) of the process's heap.
 
+   .. availability:: Linux >= 1.3.0
```
Comment (ronaldoussoren) [inline Doc/library/resource.rst]: Likewise.

File: Doc/library/resource.rst [hunk 5]
```diff
@@ -131,37 +133,51 @@ platform.
...
+
 
 .. data:: RLIMIT_STACK
 
    The maximum size (in bytes) of the call stack for the current process.  This only
    affects the stack of the main thread in a multi-threaded process.
 
+   .. availability:: Linux >= 1.3.0
```
Comment (ronaldoussoren) [inline Doc/library/resource.rst]: Likewise

File: Doc/library/resource.rst [hunk 6]
```diff
@@ -131,37 +133,51 @@ platform.
...
+
 
 .. data:: RLIMIT_RSS
 
    The maximum resident set size that should be made available to the process.
 
+   .. availability:: Linux >= 1.3.0
```
Comment (ronaldoussoren) [inline Doc/library/resource.rst]: Also available on (at least) macOS and FreeBSD, not mentioned in the unix spec.

File: Doc/library/resource.rst [hunk 7]
```diff
@@ -131,37 +133,51 @@ platform.
...
+
 
 .. data:: RLIMIT_NPROC
 
    The maximum number of processes the current process may create.
 
+   .. availability:: Linux >= 1.3.0
```
Comment (ronaldoussoren) [inline Doc/library/resource.rst]: Also available on (at least) macOS and FreeBSD, not mentioned in the unix spec.

File: Doc/library/resource.rst [hunk 8]
```diff
@@ -131,37 +133,51 @@ platform.
...
+
 
 .. data:: RLIMIT_NOFILE
 
    The maximum number of open file descriptors for the current process.
 
+   .. availability:: Linux >= 1.3.0
```
Comment (ronaldoussoren) [inline Doc/library/resource.rst]: Part of the unix spec, see above

File: Doc/library/resource.rst [hunk 9]
```diff
@@ -172,16 +188,22 @@ platform.
...
    The largest area of mapped memory which the process may occupy.
 
+   .. availability:: FreeBSD >= 11.
+
 
 .. data:: RLIMIT_AS
 
    The maximum area (in bytes) of address space which may be taken by the process.
 
+   .. availability:: Linux >= 1.3.0
```
Comment (ronaldoussoren) [inline Doc/library/resource.rst]: Part of the unix spec, see above.

**Review by ronaldoussoren — CHANGES_REQUESTED** [review]
I've added some comments about the availability of a number of constants that have much wider availability than claimed by this PR.

I have not checked beyond macOS and FreeBSD, further resource is needed to update the availability statement. The constants I marked as being available on macOS and FreeBSD as well may well be available on all relevant unix-y platforms.

The symbols I marked as being part of the unix spec should IMHO not have availability information.

**Comment by Privat33r-dev** [comment]
Issues are addressed, thanks for the review.

**Review by ronaldoussoren — APPROVED** [review]
LGTM

**Pull request merged**